src/GroupDetails.ts

```diff
@@ -0,0 +1,205 @@
+import { Group, GroupHeading } from './Group';
+import type { Grouping } from './Query';
+import type { Task } from './Task';
+
+/*
+ * This file contains implementation details of Group.ts
```

Comment:
> Then why is the code here and not in `Group.ts`? Should the code be split otherwise so that smaller classes emerge?

It was in a separate class to keep Group to a manageable size, and initially to allow me to write until tests whilst developing the behaviour, as the feedback loop of building the plugin and testing in Obsidian was too slow.

I do agree it's confusing now though.

> Right now, you also have a circular dependency between the two.

By moving all the classes, I *think* that the cycling dependency has gone.